src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java

```diff
@@ -45,6 +58,28 @@ public SecretManagerIPCAgent(SecretManager secretManager, AuthorizationHandler a
         this.authorizationHandler = authorizationHandler;
     }
 
+    /**
+     * Use secret manager config to set the cloud call thread pool size. If not configured, default value is used.
+     *
+     * @param config component configuration
+     */
+    public void setCloudCallThreadPool(Topics config) {
+        int cloudCallQueueSize = Coerce.toInt(config.lookupTopics(CONFIGURATION_CONFIG_KEY, PERFORMANCE_TOPIC)
+                .findOrDefault(DEFAULT_CLOUD_REQUEST_SIZE, CLOUD_REQUEST_QUEUE_SIZE_TOPIC));
+        if (cloudCallQueueSize <= 0) {
+            logger.atWarn().kv("queueSize", DEFAULT_CLOUD_REQUEST_SIZE)
+                    .log("Using default value for cloud request " + "queue size as the configured value is invalid");
+            cloudCallQueueSize = DEFAULT_CLOUD_REQUEST_SIZE;
+        }
+        synchronized (this) {
+            if (cloudCallThreadPool != null) {
+                cloudCallThreadPool.shutdownNow();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Call `shutdown()` to reject any incoming tasks and `awaitTermination()` to wait for existing tasks to complete or until a timeout is reached. To stop the execution and processing of tasks, if necessary, call `shutdownNow()`. Check the return value of `awaitTermination()` and handle when the method times out while waiting for other threads to stop executing. See the following diffs on GitHub that make this code change: [diff1](https://github.com/glowroot/glowroot/commit/81de5a3699c59e0cf98d485c380c1034e6cd28a5#diff-c608d2c40790b0e0e5228e6647f69729caa756bbe9127169139406314b80897eL143), [diff2](https://github.com/apache/directory-kerby/commit/3ed0e7cd3b8a0c3ce29293153e4e72f422f0637f#diff-eb42f072849f520563ec2a686789a836195f18075cc222ef355c6ea710b0a360L84)

Learn more: [ExecutorService in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html)